sass/components/html/file--link.html

```diff
@@ -1,8 +1,11 @@
 <div class="file file--link">
-  <span class="file__icon icon icon--file-link"></span>
```

Comment:
This is questionable in this specific case, because we are speaking about a modifier that might be "manually" applied sometimes and it should be only based on that, i mean if you have the file--link in a wrapper for your linked file it will have this additional styling, i would say no additional markup would be required.
From this point of view i'm more concerned about the file__title class, i think we could make it "optional" for the file--link, meaning that we could @extend the file__title selector in the <a> tag in the file--link, so that it would work even without having that element specified.
Still we should check how do we output file--link(s) at the moment and eventually simplify the markup following this new implementation.